WordPress/Classes/Models/WPAccount+Lookup.swift

```diff
@@ -0,0 +1,157 @@
+import CoreData
+
+public extension WPAccount {
+
+    // MARK: - Relationship Lookups
+
+    /// Is this `WPAccount` object the default WordPress.com account?
+    ///
+    @objc
+    var isDefaultWordPressComAccount: Bool {
+        guard let uuid = UserSettings.defaultDotComUUID, uuid.count > 0 else {
+            return false
+        }
+
+        return self.uuid == uuid
+    }
+
+    /// Does this `WPAccount` object have any associated blogs?
+    ///
+    @objc
+    var hasBlogs: Bool {
+        return !blogs.isEmpty
+    }
+
+    // MARK: - Object Lookups
+
+    /// Returns the default WordPress.com account, if one exists
+    ///
+    /// The default WordPress.com account is the one used for Reader and Notifications.
+    ///
+    static func lookupDefaultWordPressComAccount(in context: NSManagedObjectContext) throws -> WPAccount? {
+        guard let uuid = UserSettings.defaultDotComUUID, uuid.count > 0 else {
+            // No account, or no default account set. Clear the defaults key.
+            UserSettings.defaultDotComUUID = nil
+            return nil
+        }
+
+        return try lookup(withUUIDString: uuid, in: context)
+    }
+
+    /// Does a default WordPress.com account exist?
+    ///
+    /// The default WordPress.com account is the one used for Reader and Notifications.
+    ///
+    static func lookupHasDefaultWordPressComAccount(in context: NSManagedObjectContext) throws -> Bool {
+        return try lookupDefaultWordPressComAccount(in: context) != nil
+    }
+
+    /// Lookup a WPAccount by its local uuid
+    ///
+    /// - Parameters:
+    ///   - uuidString: The UUID (in string form) associated with the account
+    ///   - context: An NSManagedObjectContext containing the `WPAccount` object with the given `uuidString`.
+    /// - Returns: The `WPAccount` object associated with the given `uuidString`, if it exists.
+    ///
+    static func lookup(withUUIDString uuidString: String, in context: NSManagedObjectContext) throws -> WPAccount? {
+        let fetchRequest = NSFetchRequest<Self>(entityName: WPAccount.entityName())
+        fetchRequest.predicate = NSPredicate(format: "uuid = %@", uuidString)
+
+        guard let defaultAccount = try context.fetch(fetchRequest).first else {
+            return nil
+        }
+
+        /// This was brought over from the `AccountService`, but can (and probably should) be moved to an accessor for the property
```

Comment:
Looks like it was introduced in https://github.com/wordpress-mobile/WordPress-iOS/commit/2110d392e4eee99e9f2608e7e892b8e510d1bee0.  Not sure if a property accessor or handling when consuming API results would be better, but I'd lean to addressing it in the service or remote service layer. 🤔